framework/framework.py

```diff
@@ -166,6 +167,20 @@ def handle_offer(self, driver, offer):
         _log.debug("All tasks are running")
         return None
 
+    def get_role(self, offer):
+        """
+        Helper method to disect an offer and determine which role the resources
+        belong to.
+        :param offer:
+        :return: A string representation of the role. This can be "*" or a string.
+        """
+        # If we find anything besides "*", use it as the role. Otherwise, just use "*"
+        role = "*"
```

Comment:
After chatting with you about how a single task can be split across roles,  I think the actual correct way to do this would be to sub-divide the *required* resources between the available resources/roles in the offer ... but with the current DC/OS implementation it would amount to the same thing as we are doing here.  So, it's probably not worth doing that, and in that case, apart from the above, LGTM.